src/Native/Runtime/arm/AllocFast.S

```diff
@@ -137,11 +137,15 @@ LEAF_ENTRY RhpNewArray, _TEXT
 LOCAL_LABEL(ArrayAlignSize):
         bic         r2, r2, #3
 
-        push        {r0-r2}
+        mov         r4, r0 // Save EEType
+        mov         r5, r1 // Save element count
+        mov         r6, r2 // Save array size
         // r0 = GetThread()
         INLINE_GETTHREAD
         mov         r3, r0
```

Comment:
It would be nice to move all these 4 mov after `RhpNewArray_RarePath` label to keep the hot path as lean as possible.